src/logic/Logic.ts

```diff
@@ -29,6 +29,10 @@ export class Logic implements ILogic {
         this.selectedCity = startCity;
     }
 
+    public setTradedQuantity(quantity: number) {
```

Comment:
Please add tests for the newly added method.

Alternatively, simply make `tradedQuantity` `public`. In this case, the data field is very simple (`tradedQuantity: number`). It does not have a complicated representation that we want to hide from its users.. setter and getter only add overhead. 

FYI: An example where getter and setter are more needed would be a graph. There are many equivalent implementations of a  graph, and if we wanted to abstract from the concrete implementation we chose, we could add several getter and setter, each for different implementations of the graph.